user/languages/_includes/modern-c-cpp-toolchain.md

```diff
@@ -0,0 +1,130 @@
+### Modern C/C++ and Toolchain Versioning
+
+If you need a more modern toolchain or more granular control over your tools, you will have to handle that yourself. Fortunately, the community has already whitelisted the [repositories](https://github.com/travis-ci/apt-source-whitelist/blob/master/ubuntu.json) and [packages](https://github.com/travis-ci/apt-package-whitelist/blob/master/ubuntu-precise) necessary for most such upgrades.
+
+#### GCC on Linux
+
+Ubuntu 12.04 shipped with GCC 4.6.3 and Ubuntu 14.04 shipped with GCC 4.8.2. If you need a more recent version (if you are working with C++11, you *must* do this; support for C++11 was not finished until GCC 4.9), you will want to follow these directions:
+
+```yaml
+matrix:
+  include:
+    - os: linux
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-4.9
+      env:
+         - MATRIX_EVAL="CC=gcc-4.9 && CXX=g++-4.9"
```

Comment:
@CaseyCarter added `CC` to each GCC `MATRIX_EVAL` statement. Also moved the entire section that I'm adding out to another file so it can be included in both `c.md` and `cpp.md`.